### ISSUE TITLE
Add Your Reaction Here

### ISSUE THREAD

**Issue by chibicode** [vercel/reactions#1]
# Reactions on this comment will be reflected on https://reactions-demo.vercel.app/

This uses Next.js’s [Incremental Static Regeneration](https://github.com/zeit/next.js/discussions/11552) feature which is supported out of the box by [Vercel’s Edge Network](https://vercel.com/edge-network).

**Comment by chibicode** [comment]
## Data Fetching Code:

 https://github.com/chibicode/reactions/blob/42e08310fdbdd370a366bc6cacd043e2636c10c6/pages/index.js#L3-L60

**Comment by martinseanhunt** [comment]
This is so cool!

**Comment by anteeek** [comment]
Damn.

**Comment by Nases** [comment]
Exactly what I needed, thanks!

**Comment by Uranbold** [comment]
I am trying to build Dynamic Routes with SSR and using getInitialProps but it takes so much times before loading and my Dynamic website takes so long and I am trying to implement it.

Got a quick question?

1: I'm using the Node Express server and integrated getInitialProps can I use the getStaticProps?
2: getStaticProps (context) doesn't reply anything from the Express server?
```

export async function getStaticProps(context) {
  console.log(context);

//Page calling
  const res = await fetch( Config['apiUrl'] + 'wp/v2/pages/40' + lang)
  const json = await res.json()

  if (res.status !== 200) {
    console.error(json)
    throw new Error('Failed to fetch API')
  }

  return {
    props: {
      page,
    },
    revalidate: 36000
  }
}
```
3: I got the Multiple requests from the API and if I use getStaticProps then it should be called only from Pages. I go so many pages and can I call it in the _app.js or _document.js

Thanks.

**Comment by amir-gorji** [comment]
Are you sure its revalidation is done every 1 second or the Github gets updated values after a delay?
I get the updated values on the page after a couple of minutes.

**Comment by lemedro** [comment]
> Are you sure its revalidation is done every 1 second or the Github gets updated values after a delay?
> I get the updated values on the page after a couple of minutes.

Probably because of GitHub's API

**Comment by pungggi** [comment]
Interstingly it updated the first time after a second and a second time after some minutes..

**Comment by msreekm** [comment]
page is not refreshing after1 sec! updated only after few mins..

**Comment by nghiepnguyen520** [comment]
> page is not refreshing after1 sec! updated only after few mins..

yes, I tested and same issue with you

**Comment by mdovn** [comment]
So the first request that trigger the incremental build will alway get the stale data?

**Comment by thisisbintang** [comment]
It's fast and absolutely static.

**Comment by Abdalhadi-Jarad** [comment]
Is it expected to be updated alone every 1 second without refreshing ? 
Because I wait 20 Minutes just watching if it will update the information and still not updating

**Comment by cobrexus** [comment]
@Abdalhadi-Jarad no, since it's a static page you have a refresh a couple of times to regenerate the page.

**Comment by jrayo** [comment]
Just in case anyone wants to know how this demo behaves with different timeouts, I made a fork and you can see it on https://reactions.jrayo.vercel.app/

**Comment by zhangfaliang** [comment]
Is it expected to be updated alone every 1 second without refreshing ?
Because I wait 20 Minutes just watching if it will update the information and still not updating

**Comment by cobrexus** [comment]
@zhangfaliang no, since it's a static page you have a refresh a couple of times to regenerate the page.

**Comment by coderinblack08** [comment]
this has got to be a record :D

**Comment by Slipromise** [comment]
hello github

**Comment by javakian** [comment]
hello, hello, yes i think so

**Comment by zed-b** [comment]
Great, but I am a little bit confused, doesn't this feature just beg for custom bots to poll the static sites periodically, since the first page served after timeout is stale?

**Comment by cobrexus** [comment]
> Great, but I am a little bit confused, doesn't this feature just beg for custom bots to poll the static sites periodically, since the first page served after timeout is stale?

@zed-b the page served whilst the timeout is stale displayed a loading or skeleton or blank page whilst the new page is being regenerated.

**Comment by BITCACH** [comment]
BITCOIN^EXCHANG^INTERNET

**Comment by mkotsollaris** [comment]
@chibicode @leerob I think I am missing something. I modified the code ([see gist](
https://gist.github.com/mkotsollaris/48c94ec98d0b6490fb6d1db9078e5690#file-index-js-L59)) to print out a given timestamp passed through `getStaticProps`. I would expect that the page should regenerate every minute, and hence my timestamp shouldn't change during refresh. However, I noticed that when I refresh the page, the timestamp is always updated. thoughts? 🤔 

Update: I just realized that during `yarn dev` getStaticProps` triggers all the time during refresh, but when running from `yarn start`, getStaticProps behaves as it should!

**Comment by amansadhwani** [comment]
why do we need to refresh the page in order to check the updated value ?

**Comment by raflesgholand** [comment]
test

**Comment by xgqfrms** [comment]
#  both links not work anymore?

https://reactions-demo.now.sh/

https://reactions-demo.vercel.app/

**Comment by GBR-422777** [comment]
<img width="164" alt="Screenshot 2022-11-25 at 9 22 44 AM" src="https://user-images.githubusercontent.com/7987292/203934585-50ca4664-de88-4194-b2d0-31b50cfed3fc.png">



![Screenshot 2022-11-25 at 9 23 07 AM](https://user-images.githubusercontent.com/7987292/203934645-cdd52a97-715e-4af8-be5e-faf61852c5a6.png)




<img width="534" alt="Screenshot 2022-11-25 at 9 23 23 AM" src="https://user-images.githubusercontent.com/7987292/203934688-72c193d7-88ed-4c75-be0a-d28896734ff8.png">

**Comment by JoshuaR503** [comment]
![Screen Shot 2023-01-05 at 12 00 16 PM](https://user-images.githubusercontent.com/38048440/210869344-65dcc107-1f66-4ff5-885a-29600da8a367.png)
![Screen Shot 2023-01-05 at 12 00 32 PM](https://user-images.githubusercontent.com/38048440/210869384-41adc7b2-ed14-417c-83f5-5aa79a8ecc9c.png)

Does anybody know why my reactions are not up to date? I visited the demo app a while ago and it worked. Yesterday I visited it again and noticed the values have not changed since then.

**Comment by Levent0z** [comment]
Still not fixed. 
<img width="143" alt="CleanShot 2023-01-18 at 21 20 43@2x" src="https://user-images.githubusercontent.com/7990043/213361767-059b5911-2598-4359-915d-4a61771a8a3f.png">

**Comment by leerob** [comment]
Our GitHub personal access token expired. We'll get a new one added. In the meantime, you can see a similar demo here: https://on-demand-isr.vercel.app/

**Comment by relaxedleaf** [comment]
Hi, do we have updates on renewing the GitHub personal access token?

**Comment by mnm1984** [comment]
hi

**Comment by htrigoso** [comment]
raa

**Comment by pauloalmeida19** [comment]
sdasdasdasdasdsada

**Comment by Peter-developer01** [comment]
Why it is still not working?

**Comment by o-az** [comment]
RIP

**Comment by aleczratiu** [comment]
Ok, the 

> Our GitHub personal access token expired. We'll get a new one added. In the meantime, you can see a similar demo here: https://on-demand-isr.vercel.app/

Seems that the star number increased, this one is working thanks

**Comment by sabeerbikba** [comment]
> ![Screen Shot 2023-01-05 at 12 00 16 PM](https://user-images.githubusercontent.com/38048440/210869344-65dcc107-1f66-4ff5-885a-29600da8a367.png) ![Screen Shot 2023-01-05 at 12 00 32 PM](https://user-images.githubusercontent.com/38048440/210869384-41adc7b2-ed14-417c-83f5-5aa79a8ecc9c.png)
> 
> Does anybody know why my reactions are not up to date? I visited the demo app a while ago and it worked. Yesterday I visited it again and noticed the values have not changed since then.

Looks like api cached

**Comment by leerob** [comment]
**FYI:** We have an up-to-date demo here https://on-demand-isr.vercel.app/

**Comment by ismailarilik** [comment]
This is #1 and the most upvoted issue in GitHub. Come on, it shouldn't be that hard to renew a token. Give me the access and I will do whatever necessary without taking one cent.

**Comment by ismailarilik** [comment]
> Just in case anyone wants to know how this demo behaves with different timeouts, I made a fork and you can see it on https://reactions.jrayo.vercel.app/

This is also not up-to-date. I have just cloned the repo and try to create a demo from it but it didn't work. For some reason, GitHub respond with a 401 for a token which has read access to issues for public repositories.

**Comment by ismailarilik** [comment]
> Just in case anyone wants to know how this demo behaves with different timeouts, I made a fork and you can see it on https://reactions.jrayo.vercel.app/

It seems that this is working as expected but somehow GitHub doesn't send the latest data or this page is not displaying the correct data. UI shows 3492 right now but API returns 3488. But never mind, it is a GitHub bug and since it is not open source I cannot try to fix it. 😎
